dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java

```diff
@@ -378,14 +390,54 @@ public void setup() throws Exception {
             LibertyClientFactory.getLibertyClient(clientName);
     }
 
-    private static String getReplacementFeature(String originalFeature, Set<String> featuresToAdd) {
+    /**
+     * Obtain the replacement for a feature for this replacement action.
+     * 
+     * The lookup uses the base feature name, which is the feature name up to
+     * the first '-' character.
+     * 
+     * If EE9 replacement is active, allow the base feature name to be replaced
+     * with an EE9 replacement.
+     * 
+     * If no replacement is located, answer null, which indicates that the feature
+     * should be removed instead of being replaced.
+     *
+     * Feature names are required to have a '-', for example, "servlet-3.1".  Null
+     * is answered for feature names which do not have a '-'.
+     *
+     * @param originalFeature The feature name which is to be replaced.
+     * @param replacementFeatures Table of replacement features.
+     *
+     * @return The replacement feature name.  Null if no replacement is available.
+     */
+    private static String getReplacementFeature(String originalFeature, Set<String> replacementFeatures) {
+        String methodName = "getReplacementFeature";
         // Example: servlet-3.1 --> servlet-4.0
-        String featureBasename = originalFeature.substring(0, originalFeature.indexOf('-') + 1); // "servlet-"
-        for (String featureToAdd : featuresToAdd)
-            if (featureToAdd.contains(featureBasename)) // "servlet-4.0".contains("servlet-")
-                return featureToAdd;
+        int dashOffset = originalFeature.indexOf('-');
+        if ( dashOffset == -1 ) {
+            Log.info(c, methodName, "Remove feature [ " + originalFeature +  " ]: No '-' was found.");
+            return null;
+        }
```

Comment:
we can remove this case, or throw an exception here. All public features should have a version component

Reply:
Ok sounds good Ill make the change to throw an exception since it could be possible that manually someone adds a feature and mistakenly leaves out the '-' so that would be a good debug as to why it failed